src/lib/waku_filter/index.node.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { expect } from "chai";
+import debug from "debug";
+
+import { makeLogFileName, NOISE_KEY_1, Nwaku } from "../../test_utils";
+import { delay } from "../../test_utils/delay";
+import { Protocols, Waku } from "../waku";
+import { WakuMessage } from "../waku_message";
+
+const log = debug("waku:test");
+
+const TestContentTopic = "/test/1/waku-filter";
+
+describe("Waku Filter", () => {
+  let waku: Waku;
+  let nwaku: Nwaku;
+
+  afterEach(async function () {
+    !!nwaku && nwaku.stop();
+    !!waku && waku.stop().catch((e) => console.log("Waku failed to stop", e));
+  });
+
+  beforeEach(async function () {
+    this.timeout(10000);
+    nwaku = new Nwaku(makeLogFileName(this));
+    await nwaku.start({ filter: true });
+    waku = await Waku.create({
+      staticNoiseKey: NOISE_KEY_1,
+      libp2p: { addresses: { listen: ["/ip4/0.0.0.0/tcp/0/ws"] } },
```

Comment:
The listen address is needed for the test case to pass, which is also why this won't work in the browser.